### ISSUE TITLE
Add possibility to use custom annotations to detect query parameters

### ISSUE THREAD

**Issue by porunov** [vojtechhabarta/typescript-generator#747]
In spring it's possible to use [HandlerMethodArgumentResolver](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/method/support/HandlerMethodArgumentResolver.html) which can help to create custom data binders for Spring controllers (https://www.baeldung.com/spring-mvc-custom-data-binder#1-custom-argument-resolver).
For example, it's possible to create some methods which look like:
```
@RequestMapping(value = "/hello/{id}/world", method = RequestMethod.GET)
public ResponseEntity<GetResponse> getMethod(@PathVariable String id, 
                                             @MyCustomQueryParameter Set<String> customQueryParameter,
                                             @MyCustomQueryParameter Integer customHeaderParameter) {
    return null;
}
```
in `HandlerMethodArgumentResolver` it's possible to describe logic of how exactly those method parameters are going to be bound.
It would be great if we could somehow specify such method parameters and be able to generate them in typescript.

**Comment by vojtechhabarta** [comment]
Hello, the problem with custom resolvers is that they can do pretty much anything. They can for example get request parameter with any name (different from Java parameter name), convert strings to lists, or even get header instead of request parameter. So I don't think it is in general possible to produce TypeScript description when we don't know what the resolver is doing.

**Comment by porunov** [comment]
> Hello, the problem with custom resolvers is that they can do pretty much anything. They can for example get request parameter with any name (different from Java parameter name), convert strings to lists, or even get header instead of request parameter. So I don't think it is in general possible to produce TypeScript description when we don't know what the resolver is doing.

I agree with you that using custom argument resolver is more challenging but at least having a possibility to declare such parameters in typescript helps to resolve the issue partially. 
I made a PR #750 which might resolve this issue (at least from what I described initially). The PR doesn't solve the issue with custom names. 
That said, I have another PR #788 which could be more effective in such situations. With this PR it will be possible to use a little bit of Reflection before the typescript method is generated and solve this issue and other similar issues when we need to have different query parameters. 
For example, with #788 it will be possible to check if a specific method has any arguments annotated with `@MyCustomQueryParameter` and if so add such query parameters to the final method (or maybe the entity body or path variables, depending on the logic). 
Let me know what you think about it
